node/src/components/block_synchronizer/global_state_synchronizer.rs

```diff
@@ -135,16 +207,26 @@ impl GlobalStateSynchronizer {
     where
         REv: From<TrieAccumulatorRequest> + Send,
     {
-        match self.request_states.entry(request.block_hash) {
-            Entry::Vacant(entry) => {
-                entry.insert(RequestState::new(request));
-                self.touch();
-            }
-            Entry::Occupied(entry) => {
-                if entry.into_mut().add_request(request) {
+        // this if let works around the borrow checker...
```

Comment:
Really? `touch` works, but `enqueue_trie_for_fetching` doesn't?

Reply:
Oh, I think I know - `touch` is after the last use of `entry`, and I've been trying to use `enqueue_trie_for_fetching` before that last use. If I reorder these calls, it works :relieved: